libraries/ESP8266WiFi/src/ESP8266WiFiSTA-WPS.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ ESP8266WiFiSTA-WPS.cpp - WiFi library for esp8266
+
+ Copyright (c) 2014 Ivan Grokhotkov. All rights reserved.
+ This file is part of the esp8266 core for Arduino environment.
+
+ This library is free software; you can redistribute it and/or
+ modify it under the terms of the GNU Lesser General Public
+ License as published by the Free Software Foundation; either
+ version 2.1 of the License, or (at your option) any later version.
+
+ This library is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ Lesser General Public License for more details.
+
+ You should have received a copy of the GNU Lesser General Public
+ License along with this library; if not, write to the Free Software
+ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+
+ Reworked on 28 Dec 2015 by Markus Sattler
+
+ */
+
+
+#include "ESP8266WiFi.h"
+#include "ESP8266WiFiGeneric.h"
+#include "ESP8266WiFiSTA.h"
+#include "coredecls.h" // disable_extra4k_at_link_time()
+
+/**
+ * WPS config
+ * so far only WPS_TYPE_PBC is supported (SDK 1.2.0)
+ * @return ok
+ */
+bool ESP8266WiFiSTAClass::beginWPSConfig(void) {
+
+    // SYS ram is used by WPS, let's configure user stack inside user's HEAP
+    disable_extra4k_at_link_time();
+
+    if(!WiFi.enableSTA(true)) {
+        // enable STA failed
+        return false;
+    }
+
+    disconnect();
+
+    DEBUGV("wps begin\n");
+
+    if(!wifi_wps_disable()) {
+        DEBUGV("wps disable failed\n");
+        return false;
+    }
+
+    // so far only WPS_TYPE_PBC is supported (SDK 1.2.0)
+    if(!wifi_wps_enable(WPS_TYPE_PBC)) {
+        DEBUGV("wps enable failed\n");
+        return false;
+    }
+
+    if(!wifi_set_wps_cb((wps_st_cb_t) &WPSStatusCB)) {
+        DEBUGV("wps cb failed\n");
+        return false;
+    }
+
+    if(!wifi_wps_start()) {
+        DEBUGV("wps start failed\n");
+        return false;
+    }
+
+    esp_yield();
+    // will return here when wifi_wps_status_cb fires
+
+    return true;
+}
+
+/**
+ * WPS callback
+ * @param status wps_cb_status
+ */
+void ESP8266WiFiSTAClass::WPSStatusCB(wps_cb_status status) {

```

Comment:
This function doesn't use any class members, and there is no user-facing usage (i.e.: is private to this file). It should be a static non-member function, or a non-member function defined in an anonymous namespace.

Reply:
It was indeed an unfinished static non-member function, I can revert that back. 

About non user-facing usage, there is this comment
`// TODO user function to get status`. 

Moving this into the class has really no other goal than to keep things tight.